projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ComponentFixture, TestBed, ComponentFixtureAutoDetect, waitForAsync } from '@angular/core/testing';
+import { ReactiveFormsModule } from "@angular/forms";
+
+import { ClarityModule } from "@clr/angular";
+import { VmwSearchComponent } from "./search.component";
+
+let comp: VmwSearchComponent;
+let fixture: ComponentFixture<VmwSearchComponent>;
+let page: PageObject;
+export class PageObject {
+
+    getSearchInputElement(): HTMLInputElement {
+        return fixture.nativeElement.querySelector(`[data-test-id="search-input"]`);
+    }
+
+    getSearchButtonElement(): HTMLInputElement {
+        return fixture.nativeElement.querySelector(`[data-test-id="search-button"]`);
+    }
+
+    getClearSearchBtn(): HTMLButtonElement {
+        return fixture.nativeElement.querySelector(`[data-test-id="clear-search-btn"]`);
+    }
+
+    getSearchIcon(): HTMLElement {
+        return fixture.nativeElement.querySelector(`[data-test-id="search-icon"]`);
+    }
+
+    getHelperText(): HTMLElement {
+        return fixture.nativeElement.querySelector('[data-test-id="search-results-text"]')
+    }
+}
+
+describe('VmwSearchComponent', () => {
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [
+                ClarityModule,
+                ReactiveFormsModule
+            ],
+            declarations: [
+                VmwSearchComponent
+            ],
+            providers: [
+                { provide: ComponentFixtureAutoDetect, useValue: true },
+            ]
+        });
+    });
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(VmwSearchComponent);
+        comp = fixture.componentInstance;
+        fixture.detectChanges();
+        page = new PageObject();
+    });
+
+    it('should create', () => {
+        expect(comp).toBeTruthy();
+        expect(page.getSearchInputElement()).toBeTruthy();
+        expect(page.getSearchIcon()).toBeTruthy();
+        expect(page.getClearSearchBtn()).toBeFalsy();
+    });
+
+    // it('should show and hide "clear search button"', waitForAsync(() => {
```

Comment:
Nice commented code :). 

I vote to delete it :).

Reply:
nit: in case we decide the several lines of commented code could help us to diverge less (or diverge in a readable/understandable way) - we may keep those

Reply:
In this case, someone commented out a test. I don't think it will help us diverge less. Generally, having commented code is just bad.

Reply:
nit:  did you comment it out, or did the authors? I doubt it was the authors (haven't seen any ngx comment leftovers so far), so it's probably a test that got broekn during transferring.
I vote for deleting it, in case it's not operational. To check if operational, you need to run it, and/or descope it for CI/CD PRs (always could add task comments so we don't forget)

Reply:
I haven't commented anything out. https://gitlab.eng.vmware.com/taurus/shared-components/-/blob/person/ikoleva/internal-components-added-cleanup-theme-reduced/projects/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts#L64

Reply:
I would say, please investigate in a timeboxed manner what is the issue with the tests. 

If it get's too heavy/time consuming, in order to unblock us, please add a TODO in the code, and put a follow up ticket in the backlog, that will be taken right after we merge all the changes in.

Reply:
I fixed the issue by updating the zone.js version as detailed here https://github.com/angular/angular/issues/45476
Please don't leave commented code lying around, it's just confusing.